go/ql/lib/semmle/go/frameworks/GoMicro.qll

```diff
@@ -0,0 +1,178 @@
+/**
+ * Provides models of the [go-micro library](https://github.com/go-micro/go-micro).
+ */
+
+import go
+private import semmle.go.security.RequestForgeryCustomizations
+
+/**
+ * Module for Go-Micro framework.
+ */
+module GoMicro {
+  /**
+   * A GoMicro server type.
+   */
+  class GoMicroServerType extends Type {
+    GoMicroServerType() { this.hasQualifiedName("go-micro.dev/v4/server", "Server") }
+  }
+
+  /**
+   * A GoMicro client type.
+   */
+  class GoMicroClientType extends Type {
+    GoMicroClientType() { this.hasQualifiedName("go-micro.dev/v4/client", "Client") }
+  }
+
+  /**
+   * A file that is generated by the protobuf compiler.
+   */
+  class ProtocGeneratedFile extends File {
+    ProtocGeneratedFile() {
+      exists(string prefix, File f |
+        prefix = this.getBaseName().regexpCapture("^(.*)\\.pb(\\.micro)?\\.go$", 1) and
+        this.getParentContainer() = f.getParentContainer()
+      )
```

Comment:
What is the purpose of `f` here? And `prefix`?

Reply:
This is a very good point 🤔. I think this is due to the code being based on [Twirp support](https://github.com/github/codeql/blob/main/go/ql/lib/semmle/go/frameworks/Twirp.qll#L15). Let me see if I can remove the code.